pina/pinn.py

```diff
@@ -171,19 +152,36 @@ def merge_tensors(tensors):  # name to be changed
         for location in locations:
             condition = self.problem.conditions[location]
 
-            pts = merge_tensors([
-                condition.location.sample(
-                    argument['n'],
-                    argument['mode'],
-                    variables=argument['variables'])
-                for argument in arguments])
-
-            self.input_pts[location] = pts  #.double()  # TODO
-            self.input_pts[location] = (
-                self.input_pts[location].to(dtype=self.dtype,
-                                            device=self.device))
-            self.input_pts[location].requires_grad_(True)
-            self.input_pts[location].retain_grad()
+            samples = tuple(condition.location.sample(
+                            argument['n'],
+                            argument['mode'],
+                            variables=argument['variables'])
+                        for argument in arguments)
+            pts = PINN.merge_tensors(samples)
+
+            # TODO
+            # pts = pts.double()
+            pts = pts.to(dtype=self.dtype, device=self.device)
+            pts.requires_grad_(True)
+            pts.retain_grad()
+
+            self.input_pts[location] = pts
+
+    @staticmethod
+    def merge_tensors(tensors):  # name to be changed
+        if tensors:
+            return reduce(PINN.merge_two_tensors, tensors[1:], tensors[0])
+        raise ValueError("Expected at least one tensor")
+
+    @staticmethod
+    def merge_two_tensors(tensor1, tensor2):
+        n1 = tensor1.shape[0]
+        n2 = tensor2.shape[0]
+
+        tensor1 = LabelTensor(tensor1.repeat(n2, 1), labels=tensor1.labels)
+        tensor2 = LabelTensor(tensor2.repeat_interleave(n1, dim=0),
+                              labels=tensor2.labels)
+        return tensor1.append(tensor2)
```

Comment:
Instead of static methods, I would prefer to put such functions in the `utils.py` module

Reply:
Fixed in fe0e5d5600ae10ea4ead3f45df2af550f4ceacd0